runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -87,34 +95,53 @@ public void mergeInto(MultiObjectSMREntry other) {
      */
     @Override
     public void deserializeBuffer(ByteBuf b, CorfuRuntime rt) {
-        super.deserializeBuffer(b, rt);
-        int numStreams = b.readInt();
-        for (int i = 0; i < numStreams; i++) {
-            UUID streamId = new UUID(b.readLong(), b.readLong());
-
-            // The MultiObjectSMREntry payload is structure as follows:
-            // LogEntry Type | number of MultiSMREntry entries | MultiSMREntry id | serialized MultiSMREntry | ...
-            // Therefore we need to unpack the MultiSMREntry entries one-by-one
-            int start = b.readerIndex();
-            MultiSMREntry.seekToEnd(b);
-            int multiSMRLen = b.readerIndex() - start;
-            b.readerIndex(start);
-            byte[] streamUpdates = new byte[multiSMRLen];
-            b.readBytes(streamUpdates);
-            streamBuffers.put(streamId, streamUpdates);
+        Timer deserializeTimer = metricRegistry.timer(METRIC_PREFIX + "deserialize");
+        try (Timer.Context context = MetricsUtils.getConditionalContext(deserializeTimer)) {
+            super.deserializeBuffer(b, rt);
+            int numStreams = b.readInt();
+            for (int i = 0; i < numStreams; i++) {
+                UUID streamId = new UUID(b.readLong(), b.readLong());
+
+                Timer streamDeserializeTimer = metricRegistry.timer(
+                        METRIC_PREFIX + "deserialize-" + streamId);
+                // The MultiObjectSMREntry payload is structure as follows:
+                // LogEntry Type | number of MultiSMREntry entries | MultiSMREntry id | serialized MultiSMREntry | ...
+                // Therefore we need to unpack the MultiSMREntry entries one-by-one
+                try (Timer.Context streamContext = MetricsUtils.getConditionalContext(streamDeserializeTimer)) {
+                    int start = b.readerIndex();
+                    MultiSMREntry.seekToEnd(b);
+                    int multiSMRLen = b.readerIndex() - start;
+                    b.readerIndex(start);
+                    byte[] streamUpdates = new byte[multiSMRLen];
+                    b.readBytes(streamUpdates);
+                    streamBuffers.put(streamId, streamUpdates);
+                }
+            }
```

Comment:
do you need to catch the exception and log it?

Reply:
No, this try block utilizes `try-with-resources`
https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html

So we don't need to close this timer explicitly and it does not affect original exception workflow.